imports/position/save-position.ts

```diff
@@ -0,0 +1,50 @@
+import { DeepClient } from "@deep-foundation/deeplinks/imports/client";
+import { PACKAGE_NAME } from "./package-name";
+
+export async function savePosition(deep: DeepClient, deviceLinkId: number, coordinates: {x: number, y: number, z: number}) {
+  const geolocationEarthTypeLinkId = await deep.id("@deep-foundation/geolocation", "Earth");
+
+  if(!deviceLinkId) {
+		throw new Error("deviceLinkId must not be 0")
+	}
+
+	const containTypeLinkId = await deep.id('@deep-foundation/core', 'Contain');
+  const xTypeLinkId = await deep.id(PACKAGE_NAME, 'X');
+  const yTypeLinkId = await deep.id(PACKAGE_NAME, 'Y');
+  const zTypeLinkId = await deep.id(PACKAGE_NAME, 'Z');
+  console.log({coordinates});
+  
+  if (coordinates?.x) {
+    const {
+      data: [{ id: xLinkId }],
+    } = await deep.insert({
```

Comment:
Reduce amount of deep network requests 

_This comment applies to all the places working like these lines of code_